### PR TITLE
Add luacheck workflow and fix warnings

### PR DESCRIPTION
**Pull request by Niklp09**
- Add luacheck workflow
- Improve .luacheckrc
- Fix lc warnings

Disclaimer: This mod's code is awful...

### CODE REVIEW

File: chisel.lua [hunk 1]
```diff
@@ -91,24 +87,23 @@ local function chisel_interact(player, pointed_thing, is_right_click)
...
-				for k,v in pairs(group) do
-					new_style = k
+				for _,v in pairs(group) do
 					new_node_name = v
 				end
 			end
 		else
 			-- Forward cycle mode
-			new_style , new_node_name = next(group,style)
+			local _
```
Comment (Niklp09) [inline chisel.lua]: This is a hack, any ideas to improve it?

Comment (OgelGames) [reply chisel.lua]: You can use `select`:
```lua
new_node_name = select(2, next(group, style))
```

File: .luacheckrc [hunk 2]
```diff
@@ -1,21 +1,14 @@
...
-    table = {fields = {"copy", "getn"}},
-
-    -- Builtin
-    "vector", "ItemStack",
-    "dump", "DIR_DELIM", "VoxelArea", "Settings",
-
-    -- MTG
-    "default", "sfinv", "creative",
+	string = {fields = {"split"}},
+	table = {fields = {"copy", "getn"}},
```
Comment (OgelGames) [inline .luacheckrc:7]: These are not used anywhere.

Comment (OgelGames) [inline .luacheckrc]: These are not used anywhere.
```suggestion
```

File: chisel.lua [hunk 3]
```diff
@@ -27,18 +27,14 @@ jonez.chisel.register_chiselable_stair_and_slab = function(node_subname, group_s
...
 
 	local pos = pointed_thing.under
-	local is_sneak = player and player:get_player_control().sneak or false
-	local player_name = player and player:get_player_name()
-
-	-- A true player is required
-	if not player_name then
-		return
-	end
+	local is_sneak = player:get_player_control().sneak or false
```
Comment (OgelGames) [inline chisel.lua]: The `or false` is redundent.
```suggestion
	local is_sneak = player:get_player_control().sneak
```

**Review by OgelGames — APPROVED** [review]

**Pull request merged**